crates/uv/src/commands/project/mod.rs

```diff
@@ -125,15 +125,25 @@ pub(crate) fn init_environment(
         Err(e) => return Err(e.into()),
     }
 
-    // TODO(konsti): Extend `VersionRequest` to support `VersionSpecifiers`.
-    let requires_python_str = requires_python.map(ToString::to_string);
-    let interpreter = Toolchain::find(
-        python.or(requires_python_str.as_deref()),
-        // Otherwise we'll try to use the venv we just deleted.
-        SystemPython::Required,
-        preview,
-        cache,
-    )?
+    let interpreter = if let Some(request) = python.map(ToolchainRequest::parse).or(requires_python
+        .map(|specifiers| ToolchainRequest::Version(VersionRequest::Range(specifiers.clone()))))
+    {
+        Toolchain::find_requested(
+            &request,
+            // Otherwise we'll try to use the venv we just deleted.
+            SystemPython::Required,
+            preview,
+            cache,
+        )
+    } else {
+        Toolchain::find(
+            None,
```

Comment:
Guess I might want to make `find` take a `Option<ToolchainRequest>` as well.

Reply:
Personally find it a little easier to read as:

```rust
let interpreter = python
    .map(ToolchainRequest::parse)
    .or(requires_python
        .map(|specifiers| ToolchainRequest::Version(VersionRequest::Range(specifiers.clone()))))
    .map_or_else(
        || {
            Toolchain::find(
                None,
                // Otherwise we'll try to use the venv we just deleted.
                SystemPython::Required,
                preview,
                cache,
            )
        },
        |request| {
            Toolchain::find_requested(
                &request,
                // Otherwise we'll try to use the venv we just deleted.
                SystemPython::Required,
                preview,
                cache,
            )
        },
    )?
    .into_interpreter();
```

But totally subjective.

Reply:
Oooh that's really nice haha but... I think I still want to refactor `Toolchain::find` to just not take strings anymore so I'll probably just merge and change it later.

Reply:
Wow...

Reply:
I want it in the release and it's after work hours! hehe

Reply:
I do really like it actually <3